worker/src/ee/usageThresholds/backfillBillingCycleAnchors.ts

```diff
@@ -0,0 +1,229 @@
+import { logger } from "@langfuse/shared/src/server";
+import { prisma } from "@langfuse/shared/src/db";
+import { parseDbOrg } from "@langfuse/shared";
+import Stripe from "stripe";
+import { env } from "../../env";
+
+const stripeClient = env.STRIPE_SECRET_KEY
+  ? new Stripe(env.STRIPE_SECRET_KEY)
+  : undefined;
+
+/**
+ * TECH DEBT: This function backfills billing cycle anchors for organizations.
+ * TODO: Remove this function once all organizations have been backfilled (target: Q2 2025)
+ *
+ * Background:
+ * - Organizations created before the billing cycle anchor feature need to be backfilled
+ * - For hobby plan users: set anchor to organization createdAt
+ * - For paid plan users: fetch from Stripe subscription and use billing_cycle_anchor
+ *
+ * This function should be removed once all existing organizations have been migrated.
+ */
+export async function backfillBillingCycleAnchors(): Promise<{
+  total: number;
+  backfilled: number;
+  errors: number;
+}> {
+  const startTime = Date.now();
+  logger.info("Starting billing cycle anchor backfill");
+
+  try {
+    // Find all organizations with null billingCycleAnchor
+    const orgsToBackfill = await prisma.organization.findMany({
+      where: {
+        billingCycleAnchor: null,
```

Comment:
i'd assume that `ALTER TABLE "organizations" ADD COLUMN     "billing_cycle_anchor" TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP` will auto-backfill to now() for all existing rows and thus this wont trigger at all

i'd suggest to add a migration.sql that
1. if no active subscription is in postgres, sets billing_cycle_anchor to created_at
2. if one is in postgres, sets it to null -> this job then can pick it up

Reply:
fixed